Source/WebCore/html/HTMLScriptElement.cpp

```diff
@@ -87,9 +88,44 @@ void HTMLScriptElement::didFinishInsertingNode()
 }
 
 // https://html.spec.whatwg.org/multipage/scripting.html#dom-script-text
-void HTMLScriptElement::setText(String&& value)
+ExceptionOr<void> HTMLScriptElement::setText(std::variant<RefPtr<TrustedScript>, String>&& value)
 {
-    setTextContent(WTFMove(value));
+    ExceptionOr<String> stringValueHolder = trustedScriptCompliantStringFromVariant(*scriptExecutionContext(), WTFMove(value), "HTMLScriptElement text"_s);
+    if (stringValueHolder.hasException())
+        return stringValueHolder.releaseException();
+
+    setTextContent(stringValueHolder.releaseReturnValue());
+    return { };
+}
+
+String HTMLScriptElement::textContentForBindings() const
+{
+    return Node::textContent();
+}
+
+ExceptionOr<void> HTMLScriptElement::setTextContentForBindings(std::optional<std::variant<RefPtr<TrustedScript>, String>>&& value)
+{
+    ExceptionOr<String> stringValueHolder = trustedScriptCompliantStringFromVariant(*scriptExecutionContext(), value ? WTFMove(*value) : emptyString(), "HTMLScriptElement textContent"_s);
+    if (stringValueHolder.hasException())
+        return stringValueHolder.releaseException();
+
+    setTextContent(stringValueHolder.releaseReturnValue());
+    return { };
```

Comment:
This part of the code could be shared with `setTextContentForBindings` if we add a private member function `ExceptionOr<void> setText(ExceptionOr<String>)`.
```
return setTextContent(trustedScriptCompliantString(*scriptExecutionContext(), value ? WTFMove(*value) : emptyString(), "HTMLScriptElement textContent"_s));
```